CMakeLists.txt

```diff
@@ -25,6 +26,20 @@ option(__INTEGRATE_OPENCL "Set to ON if building LightGBM with the OpenCL ICD Lo
 
 cmake_minimum_required(VERSION 3.28)
 
+if(__INTEGRATE_OPENCL)
+  cmake_minimum_required(VERSION 3.11)
+elseif(USE_SWIG)
+  cmake_minimum_required(VERSION 3.8)
+elseif(USE_GPU OR APPLE)
+  cmake_minimum_required(VERSION 3.2)
+elseif(USE_CUDA)
+  cmake_minimum_required(VERSION 3.16)
+elseif(USE_ROCM)
+    cmake_minimum_required(VERSION 3.21)
+else()
+  cmake_minimum_required(VERSION 3.0)
+endif()
+
```

Comment:
```suggestion
```

This looks like the result of a mistake in merge conflict resolution, please remove it. The current minimum of 3.28 is higher than all of these conditions, so this is not necessary.